framework/lib/test_generation/generation/bin/evosuite.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+#
+# Wrapper script for Evosuite
+#
+# Exported environment variables:
+# D4J_HOME:                The root directory of the used Defects4J installation.
+# D4J_FILE_TARGET_CLASSES: File that lists all target classes (one per line).
+# D4J_FILE_ALL_CLASSES:    File that lists all relevant classes (one per line).
+# D4J_DIR_OUTPUT:          Directory to which the generated test suite sources
+#                          should be written (may not exist).
+# D4J_DIR_WORKDIR:         Defects4J working directory of the checked-out
+#                          project version.
+# D4J_DIR_TESTGEN_LIB:     Directory that provides the libraries of all
+#                          testgeneration tools.
+# D4J_CLASS_BUDGET:        The budget (in seconds) that the tool should spend at
+#                          most per target class.
+# D4J_SEED:                The random seed.
+
+# Check whether the D4J_DIR_TESTGEN_LIB variable is set
+if [ -z "$D4J_DIR_TESTGEN_LIB" ]; then
+    echo "Variable D4J_DIR_TESTGEN_LIB not set!"
+    exit 1
+fi
+
+# General helper functions
+source $D4J_DIR_TESTGEN_LIB/bin/_tool.source
+
+for class in $(cat $D4J_FILE_TARGET_CLASSES); do
```

Comment:
The `framework/bin/gen_tests.pl` script says that:

> Tests are generated for all classes touched by at least one triggering test.

however in here EvoSuite is only invoked for modified classes... As I commented in the `framework/bin/gen_tests.pl` script, I think the `gen_tests` module should (by default) generate test cases for all modified classes or allow people to provide a custom list of classes.

Reply:
Yes. By default, I would have test generators target only the modified classes, and allow a custom list as an option. Most of the test gen literature on D4J targets only modified classes to control cost.

Reply:
It's true that is done when running Evosuite, but I find it artificial and unrealistic.  A programmer wants tests for their entire program, not just part of it (and doesn't know the part relevant to one particular bug).

Reply:
Targeted test generation does make sense for regression testing (fixed version); I agree that it is not realistic when generating error-revealing tests (buggy version).

I updated the script to generate tests for the modified classes by default (consistent with the prior implementation), but a user can provide a set of classes that the test generator should target.

What is currently missing is the notion of there are two different use cases: generating regression tests vs. generating error-revealing tests. I suggest that we tackle this once this general test-gen interface is merged.

Reply:
I added the two test modes.